src/test/run-pass/hygiene-optout-1.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(decl_macro)]
+
+macro m($mod_name:ident) {
+    pub mod #$mod_name {
```

Comment:
The decision (https://github.com/rust-lang/rust/pull/47992#issuecomment-364309532) seems to make sure (with tests) that both `#$` and `$#` are errors because we can't currently assign them correct hygiene.
We can parse them, but then report non-fatal errors.

Reply:
Yeah, agreed.

Reply:
Coming back to this, what's the problem with `#$`? I think that's pretty easy to assign correct hygiene, and it works in my current implementation. I agree `$#` needs to be an error for now though.